src/main/java/org/iitbact/erp/services/FacilityServices.java

```diff
@@ -78,10 +80,86 @@ public BooleanResponse addFacilityData(int facilityId, FacilityRequest request)
 		BooleanResponse returnVal=new BooleanResponse(true);	
 		return returnVal;
 	}
-
-	public ErpUserProfile fetchFacilityData(FlexibleRequest request) {
-		// TODO Auto-generated method stub
-		return null;
+	
+	public BooleanResponse addFacilityAssets(int facilityId, FacilityAsset request) throws JsonProcessingException {
+		if (!this.authenticateUser(request.getAuthToken())) {
+			//TODO: Raise some error
+		}
+		Facility facility = facilityRepository.findByFacilityId(facilityId);
+		Object data = request.getData();
+		if (data != null) {
+			facility.getFacilityAsset().setData(data);
+		}
+		facilityRepository.save(facility);
+		BooleanResponse returnVal=new BooleanResponse(true);	
+		return returnVal;
+	}
+	
+	public BooleanResponse addFacilityInventory(int facilityId, FacilityInventory request) throws JsonProcessingException {
+		if (!this.authenticateUser(request.getAuthToken())) {
+			//TODO: Raise some error
+		}
+		Facility facility = facilityRepository.findByFacilityId(facilityId);
+		Object data = request.getData();
+		if (data != null) {
+			facility.getFacilityInventory().setData(data);
+		}
+		facilityRepository.save(facility);
+		BooleanResponse returnVal=new BooleanResponse(true);	
+		return returnVal;
+	}
+	public BooleanResponse addFacilityMedstaff(int facilityId, FacilityMedstaff request) throws JsonProcessingException {
+		if (!this.authenticateUser(request.getAuthToken())) {
+			//TODO: Raise some error
+		}
+		Facility facility = facilityRepository.findByFacilityId(facilityId);
+		Object data = request.getData();
+		if (data != null) {
```

Comment:
Request will silently fail, we should percolate the error upto the client. Add karne waale staff ko pata rahega he/she should retry.

Reply:
replicates elsewhere too